airbyte-cdk/python/airbyte_cdk/sources/abstract_source.py

```diff
@@ -357,3 +366,7 @@ def _get_message(self, record_data_or_message: Union[StreamData, AirbyteMessage]
             return record_data_or_message
         else:
             return stream_data_to_airbyte_message(stream.name, record_data_or_message, stream.transformer, stream.get_json_schema())
+
+    @property
+    def message_repository(self) -> Union[None, MessageRepository]:
```

Comment:
the only class extending `Source` is `AbstractSource` so I think it would make sense to either make `Source` an interface or combine them. Not something that has to be done as part of this PR though.